server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -94,6 +95,7 @@ public synchronized void updateSegments(final SegmentInfos infos, long seqNo) th
             rollTranslogGeneration();
         }
         localCheckpointTracker.fastForwardProcessedSeqNo(seqNo);
+        readerManager.maybeRefresh();
```

Comment:
Do you need to call `maybeRefresh` here? `readerManager.updateSegments(infos)` calls `maybeRefresh()` inside